### DISCUSSION TITLE
v2.4.0

### DISCUSSION THREAD

**Discussion by lukevella** [lukevella/rallly#568] [Announcements]
## 🚨 Suspending arm64 docker builds

As of this release we are suspending arm64 builds of the public docker image `lukevella/rallly`. These builds were very time consuming and often times failed due to how slow they were. If you are not running Rallly on a machine with an ARM architecture this won't affect you. If you are running on an ARM machine I would recommend pinning your version to `lukevella/rallly:latest-arm` to get the latest image that supports your architecture. Alternatively you can build the image yourself from the source code.

## 🔒 Added new security options for self-hosters

This version introduces the following environment variables you can use to secure your instance from unauthorized users.

|Name| Default | Description
|---| ---| --- |
| `AUTH_REQUIRED` | `false` | Set to `true` to require authentication for creating new polls and accessing admin pages |
| `ALLOWED_EMAILS` | `""` |  Comma separated list of email addresses that are allowed to register and login.<br>If not set, all emails are allowed.<br><br>Wildcard characters are supported.<br><br>Example: `"user@email.com, *@example.com, *@*.example.com"` |
| `DISABLE_LANDING_PAGE` | `false` | Set to `true` to take users straight to the app.<br><br>As of this version, it will take you to the poll creation page but this will change once we have an app dashboard.    |

## 🪵 Changelog
* 🔓 Add config to secure instance from unauth users by @lukevella in https://github.com/lukevella/rallly/pull/559
* 👷‍♂️ Update build command by @lukevella in https://github.com/lukevella/rallly/pull/560
* 💦 Temporarily disable arm64 builds by @lukevella in https://github.com/lukevella/rallly/pull/561
* 🌐 Update translations by @lukevella in https://github.com/lukevella/rallly/pull/558
* 💄 Hide comment dropdown menu when unavailable to user by @jonas-hoebenreich in https://github.com/lukevella/rallly/pull/564
* ⚰️ Remove dead verification code by @lukevella in https://github.com/lukevella/rallly/pull/566
* 💬 Update email templates by @lukevella in https://github.com/lukevella/rallly/pull/562
* 🌐 Update translations by @lukevella in https://github.com/lukevella/rallly/pull/565

**Full Changelog**: https://github.com/lukevella/rallly/compare/v2.3.1...v2.4.0

<hr /><em>This discussion was created from the release <a href='https://github.com/lukevella/rallly/releases/tag/v2.4.0'>v2.4.0</a>.</em>

**Comment by Kjeldgaard** [comment]
Hi,
Just to be sure, the new environment file options (AUTH_REQUIRED, ALLOWED_EMAILS, DISABLE_LANDING_PAGE) are not available using the ARM64 build, right?

**Reply by lukevella** [reply]
Unfortunately not. I know these are useful and it's unfortunate that this was round the time we were having problems building the arm64 image. I'll come back to this at some point to see if we can solve this by running a separate workflow for arm64 builds. Ideally we would use arm runners such as the ones offered by [buildjet](https://buildjet.com/) but this would be at a cost.

**Comment by Kjeldgaard** [comment]
Ok, thanks for the confirmation. Keep up the good work on Rallly 👍
